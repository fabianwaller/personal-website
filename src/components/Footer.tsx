import { Mail } from "lucide-react";
import { FaXTwitter as Twitter, FaGithub as Github, FaLinkedin as LinkedIn } from "react-icons/fa6"
import Container from "./Container";
import Logo from "./logo"
import Link from "next/link";

const Footer: React.FC = () => {
    return (
        <footer className="mt-8 relative">
            <div className="pt-8 pb-12 border-solid border-t-1 rounded-t-xl">
                <Container className="gap-y-14 grid">

                    <div className="footer__logobox">
                        <Logo />
                    </div>

                    <div className="grid grid-cols-2-max gap-x-16">
                        <ul className="flex flex-col gap-y-4">
                            <span className="block font-semibold mb-4">Links</span>
                            <li><a href="/about" className='footer__link'>About</a></li>
                            <li><a href="/journey" className='footer__link'>Journey</a></li>
                            <li><a href="/projects" className='footer__link'>Projects</a></li>
                        </ul>

                        <ul className="flex flex-col gap-y-4">
                            <span className="block font-semibold mb-4">Explore</span>
                            <li><a href="/blog" className='footer__link'>Blog</a></li>
                            {/* <li><a href="/newsletter" className='footer__link'>Newsletter</a></li> */}
                        </ul>
                    </div>

                    <div className="text-xl flex gap-6">
                        <Link href="/contact">
                            <Mail className="h-8 w-8" />
                        </Link>
                        <Link href="https://github.com/fabianwaller" target="_blank" >
                            <Github className="h-8 w-8" />
                        </Link>
                        <Link href="https://twitter.com/fabianwallerr" target="_blank" >
                            <Twitter className="h-8 w-8" />
                        </Link>
                        <Link href="https://www.linkedin.com/in/fabian-waller-974840213/" target="_blank">
                            <LinkedIn className="h-8 w-8" />
                        </Link>
                    </div>

                    <p className="text-sm text-left text-text-light">&#169; Fabian Waller. All rights reserved.</p>
                </Container>
            </div>
        </footer>
    )
}
export default Footer;