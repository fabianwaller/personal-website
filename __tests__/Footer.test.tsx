import Footer from "@/components/Footer";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";

describe("Footer", () => {
  it("renders Footer unchanged", () => {
    const component = render(<Footer />);
    expect(component.asFragment()).toMatchSnapshot();
  });
});
