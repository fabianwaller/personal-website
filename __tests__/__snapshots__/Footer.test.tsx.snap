// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders Footer unchanged 1`] = `
<DocumentFragment>
  <footer
    class="relative mt-8"
  >
    <div
      class="rounded-t-xl border-t-1 border-solid py-16"
    >
      <div
        class="container relative ml-auto mr-auto w-screen max-w-custom px-8 md:px-0 grid gap-y-14"
      >
        <div
          class="footer__logobox"
        >
          <a
            href="/"
          >
            <div
              class="flex items-center text-title-normal"
            >
              <svg
                class="mr-1 h-8 w-8"
                fill="none"
                height="74"
                viewBox="0 0 78 74"
                width="78"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class="fill-primary"
                  d="M43.1746 30.0684L68.5683 47.8492V47.8492C58.7482 61.8738 39.4183 65.2822 25.3937 55.4621V55.4621L43.1746 30.0684Z"
                />
                <path
                  class="fill-primary"
                  d="M13.1066 46.8583C17.8582 40.0722 27.2114 38.423 33.9975 43.1747V43.1747L25.3938 55.462L13.1066 46.8583V46.8583Z"
                />
                <path
                  class="fill-primary-foreground"
                  d="M33.9973 43.1746L8.6036 25.3937V25.3937C18.4237 11.3691 37.7536 7.96074 51.7782 17.7808V17.7808L33.9973 43.1746Z"
                />
                <path
                  class="fill-primary-foreground"
                  d="M64.0656 26.3848C59.3139 33.1709 49.9607 34.8201 43.1746 30.0685V30.0685L51.7783 17.7812L64.0656 26.3848V26.3848Z"
                />
              </svg>
              <span
                class="text-md font-semibold hover:text-primary"
              >
                Fabian
              </span>
            </div>
          </a>
        </div>
        <div
          class="grid grid-cols-2-max gap-x-16"
        >
          <div
            class="flex flex-col gap-y-4"
          >
            <span
              class="mb-4 block text-lg font-semibold"
            >
              Links
            </span>
            <a
              href="/"
            >
              <span
                class="hover:text-primary hover:underline"
              >
                Home
              </span>
            </a>
            <a
              href="/about"
            >
              <span
                class="hover:text-primary hover:underline"
              >
                About
              </span>
            </a>
            <a
              href="/uses"
            >
              <span
                class="hover:text-primary hover:underline"
              >
                Uses
              </span>
            </a>
            <a
              href="/projects"
            >
              <span
                class="hover:text-primary hover:underline"
              >
                Projects
              </span>
            </a>
            <a
              href="/blog"
            >
              <span
                class="hover:text-primary hover:underline"
              >
                Blog
              </span>
            </a>
            <a
              href="/contact"
            >
              <span
                class="hover:text-primary hover:underline"
              >
                Contact
              </span>
            </a>
          </div>
        </div>
        <div
          class="flex gap-6 text-xl"
        >
          <a
            class="h-8 w-8 hover:text-primary"
            href="https://x.com/fabianwallerr"
            target="_blank"
          >
            <svg
              class="h-full w-full"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M389.2 48h70.6L305.6 224.2 487 464H345L233.7 318.6 106.5 464H35.8L200.7 275.5 26.8 48H172.4L272.9 180.9 389.2 48zM364.4 421.8h39.1L151.1 88h-42L364.4 421.8z"
              />
            </svg>
          </a>
          <a
            class="h-8 w-8 hover:text-primary"
            href="https://www.instagram.com/fabianwallerr/"
            target="_blank"
          >
            <svg
              class="h-full w-full"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
              />
            </svg>
          </a>
          <a
            class="h-8 w-8 hover:text-primary"
            href="https://github.com/fabianwaller"
            target="_blank"
          >
            <svg
              class="h-full w-full"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 496 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
              />
            </svg>
          </a>
          <a
            class="h-8 w-8 hover:text-primary"
            href="https://www.linkedin.com/in/fabian-waller-974840213/"
            target="_blank"
          >
            <svg
              class="h-full w-full"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
              />
            </svg>
          </a>
          <a
            class="h-8 w-8 hover:text-primary"
            href="/rss"
            target="_blank"
          >
            <svg
              class="lucide lucide-rss h-full w-full"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 11a9 9 0 0 1 9 9"
              />
              <path
                d="M4 4a16 16 0 0 1 16 16"
              />
              <circle
                cx="5"
                cy="19"
                r="1"
              />
            </svg>
          </a>
        </div>
        <span
          class="text-left text-sm text-text-light"
        >
          Â© Fabian Waller. All rights reserved.
        </span>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
